<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marnin Wolfe" />


<title>Impute EMBRAPA GS C2 (DCas20_5360)</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">EMBRAPA_2020GS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Imputation 2019
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="convertDCas19_4403_ToVCF_102419.html">Conversion DArT report to VCF</a>
    </li>
    <li>
      <a href="Verify_gbs2dart_sampleMatches_EMBRAPA_102419.html">Verify GBS-to-DArTseqLD sample matches</a>
    </li>
    <li>
      <a href="Imputation_EMBRAPA_102419.html">Imputation steps</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Imputation 2020
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="convertDCas20_5360_ToVCF.html">Conversion DArT report DCas20_5360 to VCF</a>
    </li>
    <li>
      <a href="ImputeDCas20_5360.html">Impute GS C2 (DCas20_5360)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Other
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="about.html">About</a>
    </li>
    <li>
      <a href="license.html">License</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/wolfemd/EMBRAPA_2020GS">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Impute EMBRAPA GS C2 (DCas20_5360)</h1>
<h4 class="author">Marnin Wolfe</h4>
<h4 class="date">2020-August-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-09-01
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>EMBRAPA_2020GS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200826code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200826)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200826code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200826)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomwolfemdEMBRAPA2020GStree6e27397c0be45b378cfb6a9c538ecd02f711b285targetblank6e27397a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/wolfemd/EMBRAPA_2020GS/tree/6e27397c0be45b378cfb6a9c538ecd02f711b285" target="_blank">6e27397</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomwolfemdEMBRAPA2020GStree6e27397c0be45b378cfb6a9c538ecd02f711b285targetblank6e27397a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/wolfemd/EMBRAPA_2020GS/tree/6e27397c0be45b378cfb6a9c538ecd02f711b285" target="_blank">6e27397</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  data/OrderAppendix_1_DCas20-5360/
    Untracked:  data/Report-DCas20-5360/
    Untracked:  workflowr_log.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/ImputeDCas20_5360.Rmd</code>) and HTML (<code>docs/ImputeDCas20_5360.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/EMBRAPA_2020GS/blob/6e27397c0be45b378cfb6a9c538ecd02f711b285/analysis/ImputeDCas20_5360.Rmd" target="_blank">6e27397</a>
</td>
<td>
wolfemd
</td>
<td>
2020-09-01
</td>
<td>
EMBRAPA C2 has been imputed. Publish report and share to cassavabase
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/EMBRAPA_2020GS/ed70a5a88793ea93d4c2dd1e53323bbf076f84d3/docs/ImputeDCas20_5360.html" target="_blank">ed70a5a</a>
</td>
<td>
wolfemd
</td>
<td>
2020-08-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/EMBRAPA_2020GS/blob/bcd83b41fd6b0dc290c25ef0eab82d306d1f24ea/analysis/ImputeDCas20_5360.Rmd" target="_blank">bcd83b4</a>
</td>
<td>
wolfemd
</td>
<td>
2020-08-26
</td>
<td>
Initial publish (prior to running imputation). Debugging will be
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/EMBRAPA_2020GS/blob/9d4dfab83e5211b61386ae5ee55aa7abff8a3b86/analysis/ImputeDCas20_5360.Rmd" target="_blank">9d4dfab</a>
</td>
<td>
wolfemd
</td>
<td>
2020-08-26
</td>
<td>
Code organized into functions. Ready to run imputation of C2?
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>DArTseqLD (DCas20-5360) arrived on Aug. 22, 2020. Contains <strong>GS C2</strong> for EMBRAPA.</p>
<p>Last year’s reference panel for imputation had ~64K SNP. The C1 progeny imputed by it had &lt;9K SNP <em>after</em> post-imputation filters.</p>
<p>Options to proceed with imputation of C2 (DCas20-5360):</p>
<ol style="list-style-type: decimal">
<li>Last year’s reference panel, without C1 (64K SNP)</li>
<li>Last year’s reference panel + C1 (reduce refpanel to sites passing post-impute filter in C1 = 9K SNP)</li>
<li>Last year’s ref. panel + C1 (include unfiltered C1 data = 64K refpanel SNP)</li>
</ol>
<p>If I can use Beagle5 or latest, should be possible (fast) to create all 3 and compare the results using PCA, prediction, correlation of kinship matrices, etc.</p>
<div id="impute-embrapa-gs-c2" class="section level1">
<h1>Impute EMBRAPA GS C2</h1>
<p>Copy the imputation reference panel from 2019 to the <code>data/</code> folder.</p>
<pre class="bash"><code>cp /home/jj332_cas/marnin/EMBRAPA_2020GS /workdir/mw489/
cp /home/jj332_cas/CassavaGenotypeData/nextgenImputation2019/ImputationEMBRAPA_102419/chr*_ImputationReferencePanel_EMBRAPA_Phased_102619.vcf.gz /workdir/mw489/EMBRAPA_2020GS/data/
cp -r /home/jj332_cas/marnin/EMBRAPA_2020GS/code /workdir/mw489/EMBRAPA_2020GS/
cp -r /home/jj332_cas/CassavaGenotypeData/CassavaGeneticMap /workdir/mw489/EMBRAPA_2020GS/data/
cp /home/jj332_cas/CassavaGenotypeData/nextgenImputation2019/ImputationEMBRAPA_102419/chr*_EMBRAPA_C1progeny_FromDCas19_4301_*  /workdir/mw489/EMBRAPA_2020GS/data/ </code></pre>
<div id="refpanel-exclude-c1" class="section level2">
<h2>RefPanel: Exclude C1</h2>
<p>Impute with <a href="https://faculty.washington.edu/browning/beagle/b5_0.html">Beagle V5.0</a>.</p>
<p>Use the “imputation reference panel” dataset from 2019, e.g. <code>chr1_ImputationReferencePanel_EMBRAPA_Phased_102619.vcf.gz</code> as reference.</p>
<p>Requires 1 large memory Cornell CBSU machine (e.g. cbsulm17; 112 cores, 512 GB RAM), running 1 chromosome at a time.</p>
<p>R functions are stored in the <code>code/</code> sub-directory. Functions sourced from e.g. <strong>imputationFunctions.R</strong> are wrappers around e.g. Beagle, and other command line programs.</p>
<pre class="r"><code>targetVCFpath&lt;-here::here(&quot;data/Report-DCas20-5360/&quot;) # location of the targetVCF
refVCFpath&lt;-here::here(&quot;data/&quot;)
mapPath&lt;-here::here(&quot;data/CassavaGeneticMap/&quot;)
outPath&lt;-here::here(&quot;output/&quot;)
samplesToExclude&lt;-here::here(&quot;data&quot;,&quot;samplesToExclude.txt&quot;)
outSuffix&lt;-&quot;DCas20_5360&quot;</code></pre>
<p>Found a sample “A.Preto.joselito” in target VCF that is already in the refpanel, so I must exclude it. Created a file <code>data/samplesToExclude.txt</code> to give to Beagle.</p>
<pre class="bash"><code>bcftools query \
  --list-samples /home/jj332_cas/CassavaGenotypeData/nextgenImputation2019/ImputationEMBRAPA_102419/chr1_ImputationReferencePanel_EMBRAPA_Phased_102619.vcf.gz &gt; /home/jj332_cas/marnin/EMBRAPA_2020GS/data/chr1_ImputationReferencePanel_EMBRAPA_Phased_102619.samples</code></pre>
<pre class="r"><code>refpanel_samples&lt;-read.table(here::here(&quot;data&quot;,&quot;chr1_ImputationReferencePanel_EMBRAPA_Phased_102619.samples&quot;), 
                             header=F, stringsAsFactors = F)$V1
target_samples&lt;-read.table(here::here(&quot;data/Report-DCas20-5360/&quot;,&quot;DCas20_5360.samples&quot;), 
                             header=F, stringsAsFactors = F)$V1
samplesToExclude&lt;-target_samples[target_samples %in% refpanel_samples]
length(samplesToExclude) # [1] 644
write.table(samplesToExclude,file=here::here(&quot;data&quot;,&quot;samplesToExclude.txt&quot;),row.names = F, col.names = F, quote = F)</code></pre>
<div id="impute-c2" class="section level3">
<h3>Impute C2</h3>
<pre class="r"><code>source(here::here(&quot;code&quot;,&quot;imputationFunctions.R&quot;))

purrr::map(1:18,~runBeagle5(targetVCF=paste0(targetVCFpath,&quot;chr&quot;,.,&quot;_DCas20_5360.vcf.gz&quot;),
                            refVCF=paste0(refVCFpath,&quot;chr&quot;,.,&quot;_ImputationReferencePanel_EMBRAPA_Phased_102619.vcf.gz&quot;),
                            mapFile=paste0(mapPath,&quot;chr&quot;,.,&quot;_cassava_cM_pred.v6_91019.map&quot;),
                            outName=paste0(outPath,&quot;chr&quot;,.,&quot;_DCas20_5360_REFimputed&quot;),
                            samplesToExclude=samplesToExclude,
                            nthreads=112))</code></pre>
<p>Clean up Beagle log files after run. Move to sub-directory <code>output/BeagleLogs/</code>.</p>
<pre class="bash"><code>cd /workdir/mw489/EMBRAPA_2020GS/output/; 
mkdir BeagleLogs;
cp *_DCas20_5360_REFimputed.log BeagleLogs/
cp -r BeagleLogs /home/jj332_cas/marnin/EMBRAPA_2020GS/output/
cp *_DCas20_5360_REFimputed* /home/jj332_cas/marnin/EMBRAPA_2020GS/output/</code></pre>
</div>
<div id="post-impute-filter" class="section level3">
<h3>Post-impute filter</h3>
<p>For now, the function will just do a fixed filter: AR2&gt;0.75 (DR2&gt;0.75 as of Beagle5.0), P_HWE&gt;1e-20, MAF&gt;0.005 [0.5%].</p>
<p>It can easily be modified in the future to include parameters to vary the filter specifications.</p>
<p>Input parameters</p>
<pre class="r"><code>#&#39; @inPath path to input VCF-to-be-filtered, can be left null if path included in @inName . Must end in &quot;/&quot;
#&#39; @inName name of input VCF file EXCLUDING file extension. Assumes .vcf.gz
#&#39; @outPath path where filtered VCF and related are to be stored.Can be left null if path included in @outName . Must end in &quot;/&quot;.
#&#39; @outName name desired for output EXCLUDING extension. Output will be .vcf.gz </code></pre>
<p>Loop to filter all 18 VCF files in parallel</p>
<pre class="r"><code>inPath&lt;-here::here(&quot;output/&quot;)
outPath&lt;-here::here(&quot;output/&quot;)
source(here::here(&quot;code&quot;,&quot;imputationFunctions.R&quot;))
require(furrr); options(mc.cores=ncores); plan(multiprocess)
future_map(1:18,~postImputeFilter(inPath=inPath,
                                  inName=paste0(&quot;chr&quot;,.,&quot;_DCas20_5360_REFimputed&quot;),
                                  outPath=outPath,
                                  outName=paste0(&quot;chr&quot;,.,&quot;_DCas20_5360_REFimputedAndFiltered&quot;)))
# [1] &quot;1068 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr1_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;437 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr2_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;559 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr3_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;207 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr4_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;783 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr5_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;447 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr6_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;238 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr7_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;409 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr8_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;474 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr9_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;243 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr10_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;272 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr11_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;361 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr12_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;393 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr13_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;755 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr14_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;712 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr15_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;280 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr16_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;674 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr17_DCas20_5360_REFimputedAndFiltered&quot;
# [1] &quot;610 sites passing filter&quot;
# [1] &quot;Filtering Complete: chr18_DCas20_5360_REFimputedAndFiltered&quot;</code></pre>
<p>Check what’s left</p>
<pre class="r"><code>purrr::map(1:18,~system(paste0(&quot;zcat &quot;,here::here(&quot;output/&quot;),&quot;chr&quot;,.,&quot;_DCas20_5360_REFimputedAndFiltered.vcf.gz | wc -l&quot;)))
# 1078
# 447
# 569
# 217
# 793
# 457
# 248
# 419
# 484
# 253
# 282
# 371
# 403
# 765
# 722
# 290
# 684
# 620</code></pre>
<pre class="bash"><code>cd /workdir/mw489/EMBRAPA_2020GS/output/;
cp *_DCas20_5360_REFimputedAndFiltered* /home/jj332_cas/marnin/EMBRAPA_2020GS/output/</code></pre>
</div>
</div>
<div id="refpanel-include-c1-all-sites" class="section level2">
<h2>RefPanel: Include C1, all sites</h2>
<div id="merge-refpanel-and-c1" class="section level3">
<h3>Merge RefPanel and C1</h3>
<pre class="r"><code>inPath&lt;-here::here(&quot;data/&quot;)
outPath&lt;-here::here(&quot;data/&quot;)
source(here::here(&quot;code&quot;,&quot;imputationFunctions.R&quot;))
require(furrr); options(mc.cores=ncores); plan(multiprocess)
future_map(1:18,~mergeVCFs(inPath=inPath,
                           inVCF1=paste0(&quot;chr&quot;,.,&quot;_ImputationReferencePanel_EMBRAPA_Phased_102619&quot;),
                           inVCF2=paste0(&quot;chr&quot;,.,&quot;_EMBRAPA_C1progeny_FromDCas19_4301_REFimputed_102619&quot;),
                           outPath=outPath,
                           outName=paste0(&quot;chr&quot;,.,&quot;_ImputationReferencePanel_C1progenyAdded_EMBRAPA&quot;)))</code></pre>
<pre class="bash"><code>cd data/ 
bcftools query --list-samples chr1_ImputationReferencePanel_C1progenyAdded_EMBRAPA.vcf.gz &gt; chr1_ImputationReferencePanel_C1progenyAdded_EMBRAPA.samples</code></pre>
<pre class="r"><code>refpanel_samples&lt;-read.table(here::here(&quot;data&quot;,&quot;chr1_ImputationReferencePanel_C1progenyAdded_EMBRAPA.samples&quot;), 
                             header=F, stringsAsFactors = F)$V1
target_samples&lt;-read.table(here::here(&quot;data/Report-DCas20-5360/&quot;,&quot;DCas20_5360.samples&quot;), 
                             header=F, stringsAsFactors = F)$V1
samplesToExclude&lt;-target_samples[target_samples %in% refpanel_samples]
length(samplesToExclude) # [1] 713
write.table(samplesToExclude,file=here::here(&quot;data&quot;,&quot;samplesToExclude_RefPanelC1Added.txt&quot;),
            row.names = F, col.names = F, quote = F)</code></pre>
</div>
<div id="impute-c2-1" class="section level3">
<h3>Impute C2</h3>
<pre class="r"><code>source(here::here(&quot;code&quot;,&quot;imputationFunctions.R&quot;))
targetVCFpath&lt;-here::here(&quot;data/Report-DCas20-5360/&quot;) # location of the targetVCF
refVCFpath&lt;-here::here(&quot;data/&quot;)
mapPath&lt;-here::here(&quot;data/CassavaGeneticMap/&quot;)
outPath&lt;-here::here(&quot;output/&quot;)
samplesToExclude&lt;-here::here(&quot;data&quot;,&quot;samplesToExclude_RefPanelC1Added.txt&quot;)

purrr::map(1:18,~runBeagle5(targetVCF=paste0(targetVCFpath,&quot;chr&quot;,.,&quot;_DCas20_5360.vcf.gz&quot;),
                            refVCF=paste0(refVCFpath,&quot;chr&quot;,.,&quot;_ImputationReferencePanel_C1progenyAdded_EMBRAPA.vcf.gz&quot;),
                            mapFile=paste0(mapPath,&quot;chr&quot;,.,&quot;_cassava_cM_pred.v6_91019.map&quot;),
                            outName=paste0(outPath,&quot;chr&quot;,.,&quot;_DCas20_5360_REFimputedWithC1unfiltered&quot;),
                            samplesToExclude=samplesToExclude,
                            nthreads=112))</code></pre>
<p>Clean up Beagle log files after run. Move to sub-directory <code>output/BeagleLogs/</code>.</p>
<pre class="bash"><code>cd /workdir/mw489/EMBRAPA_2020GS/output/; 
cp *_DCas20_5360_REFimputedWithC1unfiltered.log BeagleLogs/;
cp -r BeagleLogs /home/jj332_cas/marnin/EMBRAPA_2020GS/output/;
cp *_DCas20_5360_REFimputedWithC1unfiltered* /home/jj332_cas/marnin/EMBRAPA_2020GS/output/;

cd /workdir/mw489/EMBRAPA_2020GS/data/;
cp *_ImputationReferencePanel_C1progenyAdded_EMBRAPA* /home/jj332_cas/marnin/EMBRAPA_2020GS/data/;
cp samplesToExclude_RefPanelC1Added.txt /home/jj332_cas/marnin/EMBRAPA_2020GS/data/;</code></pre>
</div>
<div id="post-impute-filter-1" class="section level3">
<h3>Post-impute filter</h3>
<pre class="r"><code>inPath&lt;-here::here(&quot;output/&quot;)
outPath&lt;-here::here(&quot;output/&quot;)
source(here::here(&quot;code&quot;,&quot;imputationFunctions.R&quot;))
require(furrr); options(mc.cores=ncores); plan(multiprocess)
future_map(1:18,~postImputeFilter(inPath=inPath,
                                  inName=paste0(&quot;chr&quot;,.,&quot;_DCas20_5360_REFimputedWithC1unfiltered&quot;),
                                  outPath=outPath,
                                  outName=paste0(&quot;chr&quot;,.,&quot;_DCas20_5360_REFimputedWithC1unfiltered_PostImputeFiltered&quot;)))</code></pre>
<pre class="r"><code>purrr::map(1:18,~system(paste0(&quot;zcat &quot;,here::here(&quot;output/&quot;),&quot;chr&quot;,.,&quot;_DCas20_5360_REFimputedWithC1unfiltered_PostImputeFiltered.vcf.gz | wc -l&quot;)))
# 1568
# 768
# 917
# 10
# 1201
# 748
# 347
# 716
# 916
# 450
# 518
# 545
# 577
# 1214
# 1179
# 406
# 1008
# 831</code></pre>
<pre class="bash"><code>cd /workdir/mw489/EMBRAPA_2020GS/output/;
cp *_DCas20_5360_REFimputedWithC1unfiltered_PostImputeFiltered* /home/jj332_cas/marnin/EMBRAPA_2020GS/output/</code></pre>
</div>
</div>
<div id="refpanel-include-c1-filtered-sites-only" class="section level2">
<h2>RefPanel: Include C1, filtered sites only</h2>
<p>Keep only the sites in the RefPanel that passed post-imputation filter for the C1</p>
<div id="filter-sites-from-refpanelc1-dataset" class="section level3">
<h3>Filter sites from RefPanel+C1 dataset</h3>
<pre class="r"><code>inPath&lt;-here::here(&quot;data/&quot;)
outPath&lt;-here::here(&quot;data/&quot;)
source(here::here(&quot;code&quot;,&quot;imputationFunctions.R&quot;))
require(furrr); options(mc.cores=18); plan(multiprocess)
furrr::future_map(1:18,~filter_positions(inPath=inPath,
                                         inVCF=paste0(&quot;chr&quot;,.,&quot;_ImputationReferencePanel_C1progenyAdded_EMBRAPA.vcf.gz&quot;),
                                         positionFile=paste0(&quot;chr&quot;,.,&quot;_EMBRAPA_C1progeny_FromDCas19_4301_REFimputed_102619.sitesPassing&quot;),
                                         outPath=outPath,
                                         outName=paste0(&quot;chr&quot;,.,&quot;_ImputationReferencePanel_C1progenyAddedFilteredSites_EMBRAPA&quot;)))</code></pre>
</div>
<div id="impute-c2-2" class="section level3">
<h3>Impute C2</h3>
<pre class="r"><code>source(here::here(&quot;code&quot;,&quot;imputationFunctions.R&quot;))
targetVCFpath&lt;-here::here(&quot;data/Report-DCas20-5360/&quot;) # location of the targetVCF
refVCFpath&lt;-here::here(&quot;data/&quot;)
mapPath&lt;-here::here(&quot;data/CassavaGeneticMap/&quot;)
outPath&lt;-here::here(&quot;output/&quot;)
samplesToExclude&lt;-here::here(&quot;data&quot;,&quot;samplesToExclude_RefPanelC1Added.txt&quot;)

purrr::map(1:18,~runBeagle5(targetVCF=paste0(targetVCFpath,&quot;chr&quot;,.,&quot;_DCas20_5360.vcf.gz&quot;),
                            refVCF=paste0(refVCFpath,&quot;chr&quot;,.,&quot;_ImputationReferencePanel_C1progenyAddedFilteredSites_EMBRAPA.vcf.gz&quot;),
                            mapFile=paste0(mapPath,&quot;chr&quot;,.,&quot;_cassava_cM_pred.v6_91019.map&quot;),
                            outName=paste0(outPath,&quot;chr&quot;,.,&quot;_DCas20_5360_REFimputedWithC1filtered&quot;),
                            samplesToExclude=samplesToExclude,
                            nthreads=112))</code></pre>
<p>Clean up Beagle log files after run. Move to sub-directory <code>output/BeagleLogs/</code>.</p>
<pre class="bash"><code>cd /workdir/mw489/EMBRAPA_2020GS/output/; 
cp *_DCas20_5360_REFimputedWithC1filtered.log BeagleLogs/;
cp -r BeagleLogs /home/jj332_cas/marnin/EMBRAPA_2020GS/output/;
cp *_DCas20_5360_REFimputedWithC1filtered* /home/jj332_cas/marnin/EMBRAPA_2020GS/output/;

cd /workdir/mw489/EMBRAPA_2020GS/data/;
cp *_ImputationReferencePanel_C1progenyAddedFilteredSites_EMBRAPA* /home/jj332_cas/marnin/EMBRAPA_2020GS/data/;
</code></pre>
</div>
<div id="post-impute-filter-2" class="section level3">
<h3>Post-impute filter</h3>
<pre class="r"><code>inPath&lt;-here::here(&quot;output/&quot;)
outPath&lt;-here::here(&quot;output/&quot;)
source(here::here(&quot;code&quot;,&quot;imputationFunctions.R&quot;))
require(furrr); options(mc.cores=ncores); plan(multiprocess)
future_map(1:18,~postImputeFilter(inPath=inPath,
                                  inName=paste0(&quot;chr&quot;,.,&quot;_DCas20_5360_REFimputedWithC1filtered&quot;),
                                  outPath=outPath,
                                  outName=paste0(&quot;chr&quot;,.,&quot;_DCas20_5360_REFimputedWithC1filtered_PostImputeFiltered&quot;)))</code></pre>
<p>Check how many sites left</p>
<pre class="r"><code>purrr::map(1:18,~system(paste0(&quot;zcat &quot;,here::here(&quot;output/&quot;),&quot;chr&quot;,.,&quot;_DCas20_5360_REFimputedWithC1filtered_PostImputeFiltered.vcf.gz | wc -l&quot;)))
# 945
# 367
# 376
# 212
# 696
# 385
# 177
# 451
# 395
# 198
# 266
# 369
# 289
# 611
# 702
# 253
# 359
# 514</code></pre>
<pre class="bash"><code>cd /workdir/mw489/EMBRAPA_2020GS/output/;
cp *_DCas20_5360_REFimputedWithC1filtered_PostImputeFiltered* /home/jj332_cas/marnin/EMBRAPA_2020GS/output/</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
